### ISSUE TITLE
Missing Carla files?

### ISSUE THREAD

**Issue by DeathHammer000** [LMMS/lmms#2543]
TLDR: CarlaNative.h and CarlaHost.h are non-existent.

I'm trying to find a way to run VSTs using Carla instead of VeSTige (which isn't working with WINE 1.8 and Qt5 for some reason). I heard that Carla does more than just VSTs so I installed the standalone package and CMake did everything from there. However, after running sudo make install, this happened:

```
[ 36%] Building CXX object plugins/carlabase/CMakeFiles/carlabase.dir/carla.o
In file included from /----/-----/Documents/lmms-1.1.3/plugins/carlabase/carla.cpp:25:0:
/----/-----/Documents/lmms-1.1.3/plugins/carlabase/carla.h:28:25: fatal error: CarlaNative.h: No such file or directory
compilation terminated.
plugins/carlabase/CMakeFiles/carlabase.dir/build.make:73: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/carla.o' failed
make[2]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/carla.o] Error 1
CMakeFiles/Makefile2:1759: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/all' failed
make[1]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

I checked for CarlaNative.h but there's no such file in either the LMMS source or the standalone Carla source. I then removed the include pointer for the file and re-ran sudo make install, but then I got this:

```
CarlaHost.h: No such file or directory
compilation terminated.
plugins/carlabase/CMakeFiles/carlabase.dir/build.make:73: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/carla.o' failed
make[2]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/carla.o] Error 1
CMakeFiles/Makefile2:1759: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/all' failed
make[1]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

I removed that, too and then I got a bunch of missing declaration errors which I can't fix. This means that there's two missing files that's preventing me from compiling with Carla. Can anyone fix this?

**Comment by falkTX** [comment]
you'll have to build and install carla first of course.

**Comment by tresf** [comment]
Closing, since it's a configuration issue with the build environment.  Please feel free to use this tracker for discussion if needed.

**Comment by DeathHammer000** [comment]
Okay, so after installing it, I'm still getting errors.

```
from /usr/local/include/carla/CarlaBackend.h:21:0,
from /usr/local/include/carla/CarlaHost.h:21,
from /----/-----/Documents/lmms-1.1.3/plugins/carlabase/carla.c
from /usr/local/include/carla/includes/CarlaDefines.h:23:21: fatal error: config.h: No such file or directory
compilation terminated.
plugins/carlabase/CMakeFiles/carlabase.dir/build.make:73: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/carla.o' failed
make[2]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/carla.o] Error 1
CMakeFiles/Makefile2:1769: recipe for target 'plugins/carlabase/CMakeFiles/carlabase.dir/all' failed
make[1]: *** [plugins/carlabase/CMakeFiles/carlabase.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

Now that I've properly installed it, what do I do? (Does make install overwrite files that were manually put there beforehand?)

**Comment by falkTX** [comment]
if you're copying the files manually, that won't work.

The carla-lmms code defines 'REAL_BUILD' which carla uses to skip the dummy config.h file.
See https://github.com/LMMS/lmms/blob/stable-1.1/plugins/carlabase/carla.cpp#L27

So if you're getting an error there you're likely doing something wrong..

**Comment by DeathHammer000** [comment]
All fixed, though I'm still having build errors (this problem hasn't been the only one; I've had to fix inline functions, misplaced operators, dependency issues and other things). Currenly WINE isn't working, but I'm almost there to full compilation and building.

(if the program doesn't work after all of that, I'll throw my laptop against the wall)
